mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1896,26 +1946,46 @@ pub fn watcher_validate_taker_payment<T: UtxoCommonOps + SwapOps>(
     coin: &T,
     input: WatcherValidatePaymentInput,
 ) -> ValidatePaymentFut<()> {
-    let mut tx: UtxoTx = try_f!(deserialize(input.payment_tx.as_slice()));
-    tx.tx_hash_algo = coin.as_ref().tx_hash_algo;
-    let first_pub = &try_f!(
+    let tx: UtxoTx = try_f!(deserialize(input.payment_tx.as_slice()));
+    let taker_pub = &try_f!(
         Public::from_slice(&input.taker_pub).map_err(|err| ValidatePaymentError::InvalidParameter(err.to_string()))
     );
-    let second_pub = &try_f!(
+    let maker_pub = &try_f!(
         Public::from_slice(&input.maker_pub).map_err(|err| ValidatePaymentError::InvalidParameter(err.to_string()))
     );
-    validate_payment(
```

Comment:
After this [comment](https://github.com/KomodoPlatform/atomicDEX-API/pull/1552#discussion_r1028257745), watcher message only includes taker payment hash instead of tx bytes to reduce size. Then the watcher retrieves the transaction with the hash. This comparison would be useful if the watcher message included the transaction bytes.

Reply:
I see, thank you!